### ISSUE TITLE
Improve multiselect dropdown in forms

### ISSUE THREAD

**Issue by davbizz** [Icinga/icingaweb2-module-director#1366]
### Problems

This is a list of things that make the multi-select drop-down difficult to use and that should be improved:

1. After click on the **Add new one** button in the drop-down if I want to see all available elements ( if I start searching I don't see all available elements ) and I click once on the down arrow nothing happens until I don't click a second time. One click should be enough in this case.
![dropdown-double-click](https://user-images.githubusercontent.com/7964685/35159030-43247cec-fd39-11e7-8c74-d0c98d3587cf.png)

2. Can't click on the right side of an already added element to add a new element instead, I have to click on an existing element. This is very confusing on the user perspective and it should be possible to click on the empty right side of the drop-down.
![click-on-right](https://user-images.githubusercontent.com/7964685/35159360-49f26948-fd3a-11e7-8f51-885f4bbd694e.png)

3. When I click on an element in the drop-down to add it the drop-down disappear but the element is not added until I don't click again on the input and then press the button start and after pressing start the full form is submitted.
![add-an-element](https://user-images.githubusercontent.com/7964685/35159612-0d2f8cce-fd3b-11e7-9f3f-178ef18bccb9.png)

4. After removing all elements from the input the input itself became unusable until I don't reload the page. 
![broken-input](https://user-images.githubusercontent.com/7964685/35159758-8d36b064-fd3b-11e7-8524-679bd8351a65.png)

5. It is difficult to distingue between selected elements and the search bar.
![selected-vs-search](https://user-images.githubusercontent.com/7964685/35159944-0fa07f08-fd3c-11e7-8850-5352d31c6de7.png)

### My Solution

An example of what is in my opinion a well-working multi-select drop-down is this:
![multi-select](https://user-images.githubusercontent.com/7964685/35160113-80d92bc0-fd3c-11e7-8dc0-a097b69524ca.png)
This dropdown has been made using [select2](https://select2.org/) and few lines of code and can be found here: https://keenthemes.com/metronic/preview/?page=components/forms/widgets/select2&demo=default

I would like to contribute in order to solve this problem and in my opinion, the best way is to substitute the existing multi-select drop-down with select2 multi-select drop-down but first I would like to know what you think about it.

**Comment by nessor** [comment]
It would be wonderful! Unfortunately, the current solution has not worked very well for me for some time. Currently no drop-down is displayed at all (I will open an issue in a moment)

**Comment by friesoft** [comment]
@nessor did you open an issue for the no dropdown problem? We are currently experiencing this as well and I couldn't find an issue for this.

**Comment by nessor** [comment]
@friesoft Yeah, I did. I have already "found" my problem. Something went wrong during installation. Machine once completely reinstalled and it works wonderfully!
